docs/versioning.rst

```diff
@@ -0,0 +1,91 @@
+.. _versioning:
+
+Vyper Versioning Guideline
+##########################
+
+Motivation
+==========
+
+Vyper has different groups that are considered "users":
+
+- Smart Contract Developers (using Vyper to write smart contracts)
+- Package Integrators (integerating Vyper package or cli into their tools)
+- Security Professionals (aware of the status of various Vyper features and security issues)
+
+The sets of users need to understand what changes require their involvement or
+understanding of how the way they use Vyper will be impacted.
+This guideline will lay out how these different groups stay informed about the progress of Vyper.
+This guideline was adapted from `semantic versioning <https://semver.org/>`_.
```

Comment:
Some of the language in this opening paragraph is a bit wordy for me. I had to read it a few times to understand. Feel free to adapt or reject this suggestion.

```suggestion
Each set of users must understand which changes to the compiler may require their
involvement, and how these changes may impact their use of the compiler.
This guide defines what scope each compiler change may have, it's potential impact based
on the type of user, and users can stay informed about the progress of Vyper.

This guide was adapted from `semantic versioning <https://semver.org/>`_.
```